dali/c_api/c_api.cc

```diff
@@ -72,17 +72,19 @@ int64_t* daliShapeAt(daliPipelineHandle* pipe_handle, int n) {
     t.ShareData(ws->Output<dali::CPUBackend>(n));
 
     std::vector<dali::Index> shape = t.shape();
-    c_shape = new int64_t[shape.size() + 1];
+    c_shape = static_cast<int64_t*>(
+      malloc(sizeof(int64_t) * (shape.size() + 1)));
```

Comment:
What is the benefit of changing to `malloc` here?

Reply:
we are passing memory ownership to the C caller. If we want the C caller to be able to free the memory we cannot use new.